src/main/java/uk/gov/companieshouse/acsp/manage/users/controller/UserAcspMembershipInternal.java

```diff
@@ -3,24 +3,67 @@
+import uk.gov.companieshouse.api.accounts.user.model.UsersList;
 import uk.gov.companieshouse.api.acsp_manage_users.api.UserAcspMembershipInternalInterface;
 import uk.gov.companieshouse.api.acsp_manage_users.model.InternalRequestBodyPatch;
 import uk.gov.companieshouse.api.acsp_manage_users.model.InternalRequestBodyPost;
 import uk.gov.companieshouse.api.acsp_manage_users.model.ResponseBodyPost;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import java.util.List;
+import java.util.Optional;
 
 @RestController
 public class UserAcspMembershipInternal implements UserAcspMembershipInternalInterface {
 
-    public UserAcspMembershipInternal() {}
+    private final AcspDataService acspDataService;
+    private final UsersService usersService;
+    private final AcspMembersRepository acspMembersRepository;
+    private final AcspMembersService acspMembersService;
+    private static final Logger LOG = LoggerFactory.getLogger(StaticPropertyUtil.APPLICATION_NAMESPACE);
+
+    @Autowired
+    public UserAcspMembershipInternal(AcspDataService acspDataService,
+                                      UsersService usersService,
+                                      AcspMembersRepository acspMembersRepository,
+                                      AcspMembersService acspMembersService) {
+        this.acspDataService = acspDataService;
+        this.usersService = usersService;
+        this.acspMembersRepository = acspMembersRepository;
+        this.acspMembersService = acspMembersService;
+    }
 
     @Override
     public ResponseEntity<ResponseBodyPost> addAcspOwner(
-            @NotNull String xRequestId,
-            @Pattern(regexp = "^[0-9A-Za-z-_]{0,32}$") String acspNumber,
-            @Valid InternalRequestBodyPost internalRequestBodyPost) {
-        return null; // TODO(https://companieshouse.atlassian.net/browse/IDVA6-1238)
+            final String xRequestId,
+            final String acspNumber,
+            final InternalRequestBodyPost internalRequestBodyPost) {
+        LOG.debugContext(xRequestId, String.format("Attempting to add ACSP owner for acsp_number %s", acspNumber), null);
+        acspDataService.fetchAcspData(acspNumber);
+        String userEmail = internalRequestBodyPost.getOwnerEmail();
+        UsersList users = Optional.ofNullable(usersService.searchUserDetails(List.of(userEmail)))
+                .filter((userList -> !userList.isEmpty()))
+                .orElseThrow(() -> {
+                    LOG.error(String.format("Failed to find user with userEmail %s", userEmail));
+                    return new NotFoundRuntimeException( "acsp-manage-users-api", "Failed to find user" );
+                });
+        String userId = users.getFirst().getUserId();
+        AcspMembersDao acspMembersDao = acspMembersService.createAcspMembersWithOwnerRole(acspNumber, userId);
+        final var acspMembers = acspMembersRepository.save(acspMembersDao);
```

Comment:
Probably just save inside of the service, would address the other comment about repository